dotnet/src/SemanticKernel.Abstractions/AI/TextCompletion/TextCompletionExtensions.cs

```diff
@@ -0,0 +1,70 @@
+
+namespace Microsoft.SemanticKernel.AI.TextCompletion;
+
+/// <summary>
+/// This sponsor class Keeps support to the previous methods to avoid any breaking changes.
+/// </summary>
+public static class TextCompletionExtensions
+{
+    /// <summary>
+    /// Creates a completion for the prompt and settings.
+    /// </summary>
+    /// <param name="textCompletion">Target interface to extend</param>
+    /// <param name="text">The prompt to complete.</param>
+    /// <param name="requestSettings">Request settings for the completion API</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Text generated by the remote model</returns>
+    public static async Task<string> CompleteAsync(this ITextCompletion textCompletion,
+        string text,
+        CompleteRequestSettings requestSettings,
+        CancellationToken cancellationToken = default)
+    {
+        var completions = await textCompletion.GetCompletionsAsync(text, requestSettings, cancellationToken).ConfigureAwait(false);
+
+        if (completions.Count == 0)
+        {
+            throw new AIException(AIException.ErrorCodes.InvalidResponseContent, "No completion result");
+        }
+
+        if (completions.Count > 1)
+        {
+            throw new NotSupportedException($"{nameof(CompleteAsync)} don't support multiple completions results. Use {nameof(ITextCompletion.GetCompletionsAsync)} instead.");
+        }
+
+        // Returns the first completion result
+        return await completions[0].CompleteAsync(cancellationToken).ConfigureAwait(false);
+    }
+
+    /// <summary>
+    /// Creates a completion for the prompt and settings.
+    /// </summary>
+    /// <param name="textCompletion">Target interface to extend</param>
+    /// <param name="text">The prompt to complete.</param>
+    /// <param name="requestSettings">Request settings for the completion API</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Streaming content of the text generated by the remote model</returns>
+    public static async IAsyncEnumerable<string> CompleteStreamAsync(this ITextCompletion textCompletion,
+        string text,
+        CompleteRequestSettings requestSettings,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        var completionResults = textCompletion.GetStreamingCompletionsAsync(text, requestSettings, cancellationToken).ConfigureAwait(false);
+
+        await foreach (var completionResult in completionResults)
+        {
+            await foreach (var word in completionResult.CompleteStreamAsync(cancellationToken))
+            {
+                yield return word;
+            }
+
+            yield return Environment.NewLine;
```

Comment:
could you explain the logic about the two loops and why we insert a new line? This will also behave differently on Windows (`/r/n`) vs Linux (`/n`), which could be a problem, e.g. with tokenization

Reply:
I'm not clearly sure why the extra line is needed, I can removed. 

I saw this in the internal implementation of Azure Clients added to their samples, I can remove it since we will have only one choice when output from the CompleteStream extension impl.

Here: [Sample04_Streaming_Chat](https://github.com/Azure/azure-sdk-for-net/blob/3fa8005a1af13f48f7bb38efa0ba272b53351ef3/sdk/openai/Azure.AI.OpenAI/tests/Samples/Sample04_StreamingChat.cs#L43)